### PR TITLE
Allow filter keys with a regex match

### PR DESCRIPTION
**Pull request by drugan**
Allows to apply a regular expression on a key instead of value. So you can get _area_ only of the _room_ entries of the _house_:

`$.house[?(/roo.*/)][area]`

https://extendsclass.com/jsonpath-tester.html

### CODE REVIEW

**Comment by Galbar** [comment]
Hello @drugan, thanks for the pull request! Sorry for the delay :sweat_smile: 

It looks great!

Could you add at least one test case for it and add it to the `JsonPath Language` section in the readme?

With that in place I'll merge this and we can have this feature in a release!

**Comment by drugan** [comment]
Hi @Galbar,

Yes, I  do it as soon as possible.

**Comment by drugan** [comment]
@Galbar 

Done. I've also added the _PCRE_ **i** and **x** modifiers which makes writing regex patterns a little easier:

https://www.php.net/manual/en/reference.pcre.pattern.modifiers.php

File: README.md [hunk 1]
```diff
@@ -96,11 +97,11 @@ JsonPath Language
 =================
 This library implements the following specification:
 ```
-var_name    = [\w\_\$^\d][\w\-\$]*
+var_name    = [filterexpr][\w\_\$^\d][\w\-\$]*
```
Comment (Galbar) [inline README.md:100]: I don't think this change was needed. With the other two I believe it's enough

```suggestion
var_name    = [\w\_\$^\d][\w\-\$]*
```

**Review by Galbar — COMMENTED** [review]
Looks good! I added a small change suggestion for the documentation.

I'll try it and run the tests (nice job, btw!) some time this week and if all goes well I'll merge it afterwards (and make a new release shortly after).

Cheers!

**Pull request merged**